packages/components/input/src/input.js

```diff
@@ -8,16 +9,38 @@ export class TSInput extends TSElement {
 
 	static get properties() {
 		return {
-			/** Disable status of input */
-			disabled: { type: Boolean, attribute: 'disabled', reflect: true }
+			/** Show error style */
+			hasError: { type: Boolean, reflect: true },
+			/** Disable state of the input */
+			disabled: { type: Boolean, reflect: true },
+			/** Readonly state of the input */
+			readonly: { type: Boolean, reflect: true },
+			/** Direction 'rtl' or 'ltr' */
+			dir: { type: String, reflect: true },
+			/** Icon that appears at the end part of the input. Readonly and disabled state will show a lock icon instead. */
+			iconRight: { type: String, reflect: true, attribute: 'icon-right' },
+			/** Icon that appears at the beginning of the input */
+			iconLeft: { type: String, reflect: true, attribute: 'icon-left' }
 		};
 	}
 
+	get direction() {
+		return this.dir ? this.dir : this.bodyDir;
+	}
+
+	get iconRightHtml() {
+		const iconRight = this.readonly || this.disabled ? 'lock' : this.iconRight;
+		return iconRight ? html` <ts-icon class="icon-right" icon=${iconRight} size="large" type="action"></ts-icon> ` : '';
```

Comment:
nit: why there are spaces at start and end? Consider to remove it.

Reply:
our format command is stupid  with lit-html stuff 😕